lib/blocktree/errors.go

```diff
@@ -20,6 +20,7 @@ var (
 	// ErrEndNodeNotFound is returned if the end of a subchain does not exist
 	ErrEndNodeNotFound = errors.New("end node does not exist")
 
+	ErrKeyNotFound = errors.New("key not found")
```

Comment:
nit should this be `ErrHashNotFound` instead? 🤔 It's a 'key' in the leaf map implementation details, but from a caller perspective, it's just a (block) hash right?

Reply:
There are multiple similar errors, there `node not found`, `end/start node not found`. I am happy with what exists right now. at least when it says key, i know that we could not find it from memory.

Reply:
Gonna skip this for now
- doesn't seem so important
- it has taken a lot of re-runs to get the CI green